src/test/extension.test.ts

```diff
@@ -1,25 +1,19 @@
 import * as assert from 'assert';
 import * as path from 'path';
 import * as vscode from 'vscode';
-import { CancellationTokenSource } from 'vscode-languageclient';
 import * as definitionProvider from '../providers/DefinitionProvider';
-import { SystemVerilogDocumentSymbolProvider } from '../providers/DocumentSymbolProvider';
-import { SystemVerilogParser } from '../parser';
-import { SystemVerilogIndexer } from '../indexer';
-import { SystemVerilogWorkspaceSymbolProvider } from '../providers/WorkspaceSymbolProvider';
 
 const examplesFolderLocation = '../../verilog-examples';
 const testFolderLocation = '../../src/test/test-files';
 
 suite('Extension Tests', () => {
     test('test #1: run build_index command', async () => {
-        // const uri = vscode.Uri.file(path.join(__dirname, testFolderLocation, 'instantiation_example.sv'));
-        // const document = await vscode.workspace.openTextDocument(uri);
-        // const editor = await vscode.window.showTextDocument(document);
-
-        setTimeout(() => {
-            vscode.commands.executeCommand('systemverilog.build_index');
+        setTimeout(async () => {
+            await vscode.commands.executeCommand('systemverilog.build_index');
         }, 500);
+        // Wait for 400ms since command execution returns
+        // before build_index is done
+        await new Promise((f) => setTimeout(f, 400));
```

Comment:
Sorry, still not following. This is what I believe will happen here:
```
time | event
-------------
0    | setTimeout()
0    | setTimeout()
400ms| f is resolved
400ms| test is completed
500ms| executeCommand is called. But maybe this is disposed because the test has ended?
```

Is the goal to run executeCommand in the background in what is effectively a speperate fork/thread?

Reply:
Maybe I'm misunderstanding also 😄 

Will not an await stall that line until the next is started? Meaning the timeout for 500ms is not started before the first is finished. However, since executeCommand returns immediately(meaning timeout never triggers), the 500ms wait is started at t=0

The idea is to trigger indexing, then to wait until it is presumably done. Problem stems from executeCommand resolving at t=0, I'm guessing it can be solved by returning a promise from build_index (but I'm not sure if Code will propagate that promise properly through execute)

Reply:
I think you want to do something like this:

```javascript
const waitFor = delay => new Promise(resolve => setTimeout(resolve, delay));

await waitFor(500);
await vscode.commands.executeCommand('systemverilog.build_index');
await waitFor(400);
```

Reply:
Good rewrite, I will change to this:
```typescript
const waitFor = delay => new Promise(resolve => setTimeout(resolve, delay));

// Trigger indexing, this returns when command is triggered and not when indexing is complete
await vscode.commands.executeCommand('systemverilog.build_index');
// Wait for indexing to (hopefully) be complete
await waitFor(400);
```